windows/src/developer/TIKE/main/Keyman.Developer.UI.UframeCEFHost.pas

```diff
@@ -308,9 +297,11 @@ procedure TframeCEFHost.Navigate;
 procedure TframeCEFHost.SetFocus;
 begin
   AssertVclThread;
-  inherited;
-  if not FIsClosing and cefwp.CanFocus then
-    cefwp.SetFocus;
+  if not FIsClosing and cefwp.CanFocus and Assigned(cef) then
```

Comment:
Is it applicable to also check `cef.CanFocus` (can it ever be false)?

Reply:
`cef: TChromium` is not a visual component in Delphi context so it doesn't have a `CanFocus` function. It is hosted by `cefwp: TCEFWindowParent` and it is `cefwp`'s responsibility to coordinate focus etc with Delphi.